Remove DASH/SHACL-JS support

martinmaillard opened the pull request:
Addresses #7 

This is quite huge and hard to review... sorry about that.

This PR removes DASH and everything related to dynamically loading validations from JS libraries. There is a lot of cleanup that will be made possible by this change, but I'll do it in separate PRs.

Some highlights of this change:
- Tests are running much faster... which probably means that the lib is faster ;)
- No more Gulp
- No more imported code from TopQuadrant, which will make it possible to slowly remove some of the complexity.

bergos commented:
The vocabs package would also add the N3 parser as dependency. That's quite heavy. I found an unfinished piece of code for a serializer to RDFJS in my file system, which I started with use cases like this in mind. Maybe you can use it and manually trigger it. There should not be to many changes in the shacl vocab. I put the code in a [gist](https://gist.github.com/bergos/417c5214c95f77e4f296b63de02c9e5b). The actual code is less than 50 lines and hopefully self-explanatory.

martinmaillard commented:
I think I understand the idea. I have an almost working solution, but some tests are failing in some cases and I think it has to do with the way blank nodes are serialized. Since the blank node ID is hard-coded in the serialized module, I think it can cause clashes when loading the data back up.

martinmaillard committed:
Extract validation functions from dash

This is the smallest change I could do to extract these functions. There
is a lot of cleanup that can be done now.

martinmaillard committed:
Refactor ConstraintComponent into a class

martinmaillard committed:
Remove DASH and javascript libraries

martinmaillard committed:
Remove rdfquery import mechanism and gulp dependency

The features in RDFQuery will be owned and modified by this library, as
the original one is not compatible with the RDFJS data-model. We don't
need to be able to import it anymore.

martinmaillard committed:
Cleanup rdfquery.js

martinmaillard committed:
Refactor rdfquery with classes

martinmaillard committed:
Fix constraint error messages

martinmaillard commented on src/validation-function.js:
```diff
@@ -63,6 +63,8 @@ class ValidationFunction {
         args.push('DummyShapesGraph')
       } else if (arg === 'this') {
         args.push(focusNode)
+      } else if (arg === 'context') {
+        args.push(this.context)
```
Injecting the context is a short-term solution to make it work. I will remove the whole dependency injection mechanism in a separate PR. Every validation function should probably take the same set of arguments. I may even put then in a class that already has access to the context.

martinmaillard commented on src/validators.js:
```diff
@@ -0,0 +1,514 @@
+// Functions implementing the validators of SHACL-JS
```
Functions in this file were directly extracted from `dash.js`. Apart from applying `standard` and injecting the `$context` in all of them, I didn't do any changes.

bergos reviewed (approved):
Yes, quite big PR, so maybe I missed some details, but it looks good in general.
